### PR TITLE
Store the tracepoint events in the client

### PR DESCRIPTION
**Pull request by MariaSandru27**
In order for the tracepoint events to recognize the names and categories of the tracepoints, they store a hash key that represents
these fields. The keys are recognized by being separately mapped in an unordered map to the corresponding name and category.

### CODE REVIEW

File: OrbitClientGgp/CMakeLists.txt [hunk 1]
```diff
@@ -7,19 +7,20 @@ project(OrbitClientGgp)
...
-target_link_libraries(OrbitClientGgp PUBLIC 
-    OrbitBase
-    OrbitCaptureClient
-    OrbitClientServices
-    OrbitProtos)
+target_link_libraries(OrbitClientGgp PUBLIC
+        OrbitBase
+        OrbitCaptureClient
+        OrbitClientServices
+        OrbitGl
```
Comment (kisenshi) [inline OrbitClientGgp/CMakeLists.txt:24]: OrbitClientGgp must be independent from OrbitGl so please do not include it, thanks!

Comment (MariaSandru27) [reply OrbitClientGgp/CMakeLists.txt:24]: I understand. Thank you!

File: OrbitGl/TracepointInfoManager.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#include <optional>
+#include <string>
+
+#include "BlockChain.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/synchronization/mutex.h"
+#include "capture_data.pb.h"
+#include "tracepoint.pb.h"
+
+class TracepointInfoManager {
```
Comment (kisenshi) [inline OrbitGl/TracepointInfoManager.h:17]: As briefly discussed, this should be created outside OrbitGl so the client ggp can make use of it, as it is independent from OrbitGl. Thanks!

**Review by kisenshi — CHANGES_REQUESTED** [review]
Thanks for adding me. I left some comments; Please let me know if you have any questions :)

File: OrbitGl/TracepointInfoManager.h (hunk 2 above)
Comment (MariaSandru27) [reply OrbitGl/TracepointInfoManager.h:17]: Thank you so much for pointing that out!

File: OrbitCore/TracepointInfoManager.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TracepointInfoManager.h"
+
+bool TracepointInfoManager::AddUniqueTracepointEventInfo(
```
Comment (kisenshi) [inline OrbitCore/TracepointInfoManager.cpp:7]: I don't really have context about what you are doing but just wondering why you return a boolean when this is called in the listener without checking what the method returns.

File: OrbitClientGgp/CMakeLists.txt [hunk 4]
```diff
@@ -7,19 +7,18 @@ project(OrbitClientGgp)
 add_executable(OrbitClientGgp main.cpp)
 
 target_compile_options(OrbitClientGgp PRIVATE
-    ${STRICT_COMPILE_FLAGS})
```
Comment (kisenshi) [inline OrbitClientGgp/CMakeLists.txt:10]: The CMakeLists.txt is not change at all now  so maybe it is not worth it to include these indentation updates in the commit

**Review by kisenshi — APPROVED** [review]

File: OrbitClientGgp/CMakeLists.txt (hunk 4 above)
Comment (MariaSandru27) [reply OrbitClientGgp/CMakeLists.txt:10]: The identation updates were not intentional but were caused by the formatting in CLion. However, I have found a way to remove the identation updates. Thanks you a lot for taking the time to review this and point this out!

File: OrbitCore/TracepointInfoManager.cpp (hunk 3 above)
Comment (MariaSandru27) [reply OrbitCore/TracepointInfoManager.cpp:7]: This is intended to be useful in the long-term for different checks. If you look at the way it is done for the string manager or the callstacks, the mechanism is similar so, I also wanted to keep it consistent with the rest of the data retrieved from the collector.

**Comment by MariaSandru27** [comment]
> I don't have the context of what you are doing so I can't really say much about the tracepoints updates themselves (althought I left some general comments) so this should be approved by someone who has that context :) However, I took a look at the changes in ClientGgp and they make sense as it is updated the same way to the Orbit App, while keep being independent from OrbitGl. Thanks a lot for updating it as well!

Thanks a lot for taking the time to review my code!;)

File: OrbitClientGgp/ClientGgp.h [hunk 5]
```diff
@@ -40,12 +41,16 @@ class ClientGgp final : public CaptureListener {
...
+  void OnUniqueTracepointInfo(uint64_t key,
+                              orbit_grpc_protos::TracepointInfo tracepoint_info) override;
+  void OnTracepointEvent(orbit_client_protos::TracepointEventInfo tracepoint_event_info) override;
 
  private:
   ClientGgpOptions options_;
   std::shared_ptr<grpc::Channel> grpc_channel_;
   std::shared_ptr<Process> target_process_;
   std::shared_ptr<StringManager> string_manager_;
+  std::shared_ptr<TracepointInfoManager> tracepoint_info_manager_;
```
Comment (danielfenner) [inline OrbitClientGgp/ClientGgp.h:53]: Is there a good reason to have TracepointInfoManager stored in a shared_ptr and not an unique_ptr?

This confused me at first (I thought OrbitApp and ClientGgp are sharing the same instance - which is not the case).

I see that others are using shared_ptr as well. So please keep as is if the shared ownership is intended and needed further down the road.

**Review by danielfenner — APPROVED** [review]
lgtm - please have a look at the comments. If you decide to keep this as is I think this is good to go.

**Comment by danielfenner** [comment]
The failed check is a flaky test that should be fixed by now. So it is save to ignore (or to push again to have it pass).

**Pull request merged**